src/scripting/lua_kernel_base.cpp

```diff
@@ -95,6 +94,63 @@ static int intf_compare_versions(lua_State* L)
 	return 1;
 }
 
+/**
+ * Decomposes a version into its component parts
+ */
+static int intf_decompose_version(lua_State* L) {
+	version_info vers(luaL_checkstring(L, 1));
+	lua_createtable(L, vers.components().size(), vers.special_version_separator() ? 5 : 3);
+	lua_pushinteger(L, vers.major_version());
+	lua_setfield(L, -2, "major");
+	lua_pushinteger(L, vers.minor_version());
+	lua_setfield(L, -2, "minor");
+	lua_pushinteger(L, vers.revision_level());
+	lua_setfield(L, -2, "revision");
+	if(char sep = vers.special_version_separator()) {
+		lua_pushlstring(L, &sep, 1);
+		lua_setfield(L, -2, "sep");
+		std::string special = vers.special_version();
+		lua_pushlstring(L, special.c_str(), special.size());
+		lua_setfield(L, -2, "special");
+	}
+	int i = 0;
+	for(int n : vers.components()) {
+		lua_pushinteger(L, n);
+		lua_seti(L, -2, ++i);
+	}
+	return 1;
+}
+
+/**
+ * Builds a version string from its component parts
+ */
+static int intf_make_version(lua_State* L) {
+	int major = luaL_checkinteger(L, 1), minor = luaL_checkinteger(L, 2), rev = luaL_checkinteger(L, 3);
+	std::string sep, special;
+	if(!lua_isnoneornil(L, 4)) {
+		special = luaL_checkstring(L, 4);
+		if(!special.empty() && std::isalpha(special[0])) {
+			sep.push_back('+');
+		} else {
+			sep.push_back(special[0]);
+			special = special.substr(1);
+		}
+	} else {
+		sep.push_back(0);
+	}
+	version_info vers(major, minor, rev, sep[0], special);
+	lua_push(L, vers.str());
+	return 1;
+}
+
+/**
+ * Returns the current Wesnoth version
+ */
+static int intf_current_version(lua_State* L) {
```

Comment:
These version functions should have unit tests.

Reply:
Obviously I can't unit test current_version though.

I guess I could copy the version tests from the Boost unit tests and convert them to Lua…

Reply:
A unit test would at least check that it doesn't crash and returns a plausible version-like string.